tools/ctl/ctl.go

```diff
@@ -1184,9 +1185,91 @@ var runAgent = &cobra.Command{
 		return nil
 	},
 }
+
+var localAgent = &cobra.Command{
+	Use:   "local-agent --project <project> --agent-api <URI> --metadata-bucket <bucket> --ecosystem <ecosystem> --package <name> --version <version> --artifact <name>",
+	Short: "Run benchmark",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := cmd.Context()
+		t := rebuild.Target{Ecosystem: rebuild.Ecosystem(*ecosystem), Package: *pkg, Version: *version, Artifact: *artifact}
+		if t.Ecosystem == "" || t.Package == "" || t.Version == "" || t.Artifact == "" {
+			log.Fatal("ecosystem, package, version, and artifact must be provided")
+		}
+		if *project == "" {
+			return errors.New("project must be provided")
+		}
+		if *agentApiUri == "" {
+			return errors.New("agent API endpoint not provided")
+		}
+		if *metadataBucket == "" {
+			return errors.New("metadata bucket not provided")
+		}
+		fire, err := firestore.NewClient(cmd.Context(), *project)
+		if err != nil {
+			return errors.Wrap(err, "creating firestore client")
+		}
+		sessionTime := time.Now().UTC()
+		sessionID := fmt.Sprintf("session-%s", sessionTime.Format(time.RFC3339))
+		// Set defaults for configuration
+		maxIterations := 10
+		session := schema.AgentSession{
+			ID:             sessionID,
+			Target:         t,
+			MaxIterations:  maxIterations,
+			TimeoutSeconds: 60 * 60, // 1 hr
+			Context:        &schema.AgentContext{},
+			Status:         schema.AgentSessionStatusRunning,
+			JobName:        "",
+			Created:        sessionTime,
+			Updated:        sessionTime,
+		}
+		// Create session in Firestore
+		err = fire.RunTransaction(ctx, func(ctx context.Context, t *firestore.Transaction) error {
+			// NOTE: This would fail if the
+			return t.Create(fire.Collection("agent_sessions").Doc(sessionID), session)
+		})
+		if err != nil {
+			if status.Code(err) == codes.AlreadyExists {
+				return errors.Errorf("agent session %s already exists", sessionID)
+			}
+			return errors.Wrap(err, "creating agent session")
+		}
+		log.Println("Created session object")
+		// Run agent locally
+		client, err := oauth.AuthorizedUserIDClient(ctx)
+		if err != nil {
+			log.Fatal(errors.Wrap(err, "creating authorized HTTP client"))
+		}
+		baseURL, err := url.Parse(*agentApiUri)
+		if err != nil {
+			log.Fatalf("Failed to parse agent API URL: %v", err)
+		}
+		// Create agent API client stubs
+		iterationStub := api.Stub[schema.AgentCreateIterationRequest, schema.AgentCreateIterationResponse](client, baseURL.JoinPath("agent/session/iteration"))
+		completeStub := api.Stub[schema.AgentCompleteRequest, schema.AgentCompleteResponse](client, baseURL.JoinPath("agent/session/complete"))
+		deps := agent.RunSessionDeps{
+			IterationStub:  iterationStub,
+			CompleteStub:   completeStub,
+			Agent:          agent.NewDefaultAgentLogic(),
```

Comment:
didn't see this in the previous PR but I'm not a fan of "logic" here. we could just call this "Agent", right?